File: 0001_reverse_string/Solution.java
public class Solution {
    public String reverseString(String s) {
        char[] arr = s.toCharArray();
        int i = 0;
        int j = s.length() - 1;
        while (i < j) {
            char t = arr[i];
            arr[i] = arr[j];
            arr[j] = t;
            i++;
            j--;
        }

        return new String(arr);
    }

    public String reverseStringByStringBuilder(String s) {
        return new StringBuilder(s).reverse().toString();
    }

    public static void main(String[] args) {
        Solution s = new Solution();
        System.out.println(s.reverseString("hello")); // olleh
        System.out.println(s.reverseStringByStringBuilder("hello")); // olleh
    }
}

File: 0001_reverse_string/solution.go
package main

import (
	"fmt"
)

func main() {
	fmt.Println(reverseString("Hello"))
}

func reverseString(s string) string {
	chars := []int32(s)

	i := 0
	j := len(chars) - 1

	for i < j {
		chars[i], chars[j] = chars[j], chars[i]
		i++
		j--
	}

	return string(chars)
}

